ui/src/data_explorer/containers/DataExplorer.tsx

```diff
@@ -60,12 +66,14 @@ export class DataExplorer extends PureComponent<Props, State> {
 
     this.state = {
       isWriteFormVisible: false,
+      isSendToDashboardVisible: false,
     }
   }
 
-  public componentDidMount() {
-    const {source, autoRefresh} = this.props
+  public async componentDidMount() {
+    const {source, autoRefresh, handleGetDashboards} = this.props
     const {query} = qs.parse(location.search, {ignoreQueryPrefix: true})
+    await handleGetDashboards()
```

Comment:
before making this request, could you check if dashboards are already in the redux store? it would be great to avoid making this request unless we absolutely have to, since it's quite expensive (it loads every cell for every dashboard as well).